specification/servicenetworking/resource-manager/Microsoft.ServiceNetworking/preview/2024-05-01-preview/TrafficController.json

```diff
@@ -0,0 +1,1930 @@
+        "value": {
+          "type": "array",
+          "description": "The TrafficController items on this page",
+          "items": {
+            "$ref": "#/definitions/TrafficController"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "TrafficControllerProperties": {
+      "type": "object",
+      "description": "Traffic Controller Properties.",
+      "properties": {
+        "configurationEndpoints": {
+          "type": "array",
+          "description": "Configuration Endpoints.",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "frontends": {
+          "type": "array",
+          "description": "Frontends References List",
+          "items": {
+            "$ref": "#/definitions/ResourceId"
+          },
+          "readOnly": true
+        },
+        "associations": {
+          "type": "array",
+          "description": "Associations References List",
+          "items": {
+            "$ref": "#/definitions/ResourceId"
+          },
+          "readOnly": true
+        },
+        "securityPolicies": {
+          "type": "array",
+          "description": "Security Policies References List ",
+          "items": {
+            "$ref": "#/definitions/ResourceId"
+          },
+          "readOnly": true
+        },
+        "securityPolicyConfigurations": {
+          "$ref": "#/definitions/SecurityPolicyConfigurations",
+          "description": "Security Policy Configuration",
+          "x-ms-identifiers": []
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "The status of the last operation.",
+          "readOnly": true
+        }
+      }
+    },
+    "TrafficControllerUpdate": {
+      "type": "object",
+      "description": "The type used for update operations of the TrafficController.",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "description": "Resource tags.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/TrafficControllerUpdateProperties",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "TrafficControllerUpdateProperties": {
+      "type": "object",
+      "description": "The updatable properties of the TrafficController.",
+      "properties": {
+        "securityPolicyConfigurations": {
+          "$ref": "#/definitions/SecurityPolicyConfigurationsUpdate",
+          "description": "Security Policy Configuration",
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "WafPolicy": {
+      "type": "object",
+      "description": "Web Application Firewall Policy",
+      "properties": {
+        "id": {
```

Comment:
>id [](http://example.com/codeflow?start=9&length=2)

The name id: conflicts with top level property name of this resource, so you need a modifier, e.g. resourceId, or webApplicationPolicyResourceId, or ...

That comment applies to other types here as well.

Reply:
@mentat9 what do you mean by a conflict? There are no properties in the top-level resource with the same name. Additionally the 'id' property is a field in several other resources that are used in current API versions.

Reply:
Discussed this issue on DM.

Reply:
This is ARM guidance. Per separate discussion I won't block signoff on using this name. Please be sure to apply this guidance on future APIs.